bazel/envoy_build_system.bzl

```diff
@@ -195,6 +196,57 @@ def envoy_include_prefix(path):
         return "/".join(path.split("/")[1:])
     return None
 
+# External CMake C++ library targets should be specified with this function. This defaults
+# to building the dependencies with ninja
+def envoy_cmake_external(
+        name,
+        cache_entries = {},
+        cmake_options = ["-GNinja"],
+        make_commands = ["ninja", "ninja install"],
+        lib_source = "",
+        postfix_script = "",
+        static_libraries = [],
+        copy_pdb = False):
+    # On Windows, we don't want to explicitly set CMAKE_BUILD_TYPE,
+    # rules_foreign_cc will figure it out for us
+    cache_entries_no_build_type = {key: cache_entries[key] for key in cache_entries.keys() if key != "CMAKE_BUILD_TYPE"}
+
+    pf = ""
+    if copy_pdb:
+        name_to_pdb = {"ares": "c-ares", "nghttp2": "nghttp2_static", "zlib": "zlibstatic"}
+        pdb_name = name_to_pdb.get(name, name)
```

Comment:
I like all of this PR except this bit :) Can we pass in the pdf_name as an arg to `envoy_cmake_external`, to avoid having target specific logic/defines in the macro?

Reply:
makes sense, done